### PR TITLE
feat(image_projection_based_fusion): unrectify image projection based fusion

### PR DESCRIPTION
**Pull request by yukkysaito**
## Description

In image_projection_based_fusion, only the internal parameters of the camera were considered, and no distortion correction was applied. As a result, when projecting onto images with significant distortion, misalignments occurred. This Pull Request has been modified to account for distortion.

## Related links

[TIER IV INTERNAL](https://tier4.atlassian.net/browse/RT1-6051)

## Tests performed

In roi_cluster_fusion, operation has been confirmed on the actual machine.
before
image

after
image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/image_projection_based_fusion/src/utils/utils.cpp [hunk 1]
```diff
@@ -13,8 +13,18 @@
...
 #include "image_projection_based_fusion/utils/utils.hpp"
+
 namespace image_projection_based_fusion
 {
+Eigen::Vector2d calcRawImageProjectedPoint(
+  const image_geometry::PinholeCameraModel & pinhole_camera_model, const cv::Point3d & point3d)
+{
+  cv::Point2d rectified_image_point = pinhole_camera_model.project3dToPixel(point3d);
+
+  cv::Point2d raw_image_point = pinhole_camera_model.unrectifyPoint(rectified_image_point);
```
Comment (kminoda) [inline perception/image_projection_based_fusion/src/utils/utils.cpp:24]:
```suggestion
  const cv::Point2d rectified_image_point = pinhole_camera_model.project3dToPixel(point3d);

  const cv::Point2d raw_image_point = pinhole_camera_model.unrectifyPoint(rectified_image_point);
```

**Review by kminoda — APPROVED** [review]

**Comment by technolojin** [comment]
@yukkysaito I checked there is a compile failure. It was caused by https://github.com/autowarefoundation/autoware.universe/pull/7224, and it was fixed.

**Comment by technolojin** [comment]
@kminoda DCO warning seems to be shown by your `Co-authored-by`. Can you resolve it?

**Comment by yukkysaito** [comment]
> @kminoda DCO warning seems to be shown by your Co-authored-by. Can you resolve it?

done

**Pull request merged**